### PR TITLE
[Core] BREAKING CHANGE: `az extension add/update` and `az upgrade`: Set default `--allow-preview` value from `True` to be `False`

### PR DESCRIPTION
**Pull request by AllyW**
**Related command**

**Description**

**Testing Guide**

**History Notes**

[Extension] BREAKING CHANGE: `az extension add/update`: Set default `--allow-preview` value from `True` to be `False`
[Upgrade] BREAKING CHANGE: `az upgrade`: Set default `--allow-preview` value from `True` to be `False`

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Core

File: scripts/ci/test_extensions.sh [hunk 1]
```diff
@@ -34,7 +34,7 @@ for ext in $output; do
     fi
 
     echo "Verifying extension:" $ext
-    az extension add -n $ext
+    az extension add -n $ext --allow-preview True
```
Comment (bebound) [inline scripts/ci/test_extensions.sh:37]: How many extensions do not follow the latest version number convention?

The test script should also use the default config. We can make an exception list for outdated extensions.

**Comment by bebound** [comment]
The PR title does not follow the convention. We need to put command name at the beginning of the title.

File: src/azure-cli-core/azure/cli/core/extension/operations.py [hunk 2]
```diff
@@ -306,12 +306,6 @@ def check_version_compatibility(azext_metadata):
 def add_extension(cmd=None, source=None, extension_name=None, index_url=None, yes=None,  # pylint: disable=unused-argument, too-many-statements
                   pip_extra_index_urls=None, pip_proxy=None, system=None,
                   version=None, cli_ctx=None, upgrade=None, allow_preview=None):
-    if allow_preview is None:
```
Comment (kairu-ms) [inline src/azure-cli-core/azure/cli/core/extension/operations.py:309]: Please add error log to instruct users add `--allow-preview` if an extension has preview version only but users forget adding `--allow-preview` argument.

**Review by kairu-ms — APPROVED** [review]

**Comment by AllyW** [comment]
After cli released, customers will get error message as below for extension modules which do not have stable versions.
image

File: scripts/ci/test_extensions.sh (hunk 1 above)
Comment (AllyW) [reply scripts/ci/test_extensions.sh:37]: The latest version schema and cli core code is compatible with the version number of all extension modules. 

There are 94 extension modules that only has preview releases (which would be failed when customers use new cli to install extension, and if it fails, the message like following will be prompted to users.) 

image

**Pull request closed without merging**